PubHub/PubHub.Common/Services/EpubReaderService.cs

```diff
@@ -0,0 +1,98 @@
+    {
+        // A thing to be aware of here, is that the first few pages can contain only little information and maybe images, which haven't been dealt with for now.
+
+        /// <summary>
+        /// Retrieves the <see cref="EpubBook"/> from the <see cref="byte[]"/> with the book content stored in the db.
+        /// </summary>
+        /// <param name="epubContent"><see cref="byte[]"/> containing the book content.</param>
+        /// <returns>A <see cref="ServiceResult{TResult}"/> with how the request was handled.</returns>
+        public async Task<ServiceResult<EpubBook>> GetEpubBookAsync(byte[] epubContent)
+        {
+            try
+            {
+                using MemoryStream stream = new MemoryStream(epubContent);
+                var epubBook = await EpubReader.ReadBookAsync(stream);
+                return new ServiceResult<EpubBook>(epubBook, errorDescriptor: "Successfully retrieved the Epub book!");
+            }
+            catch (Exception ex)
+            {
+                return new ServiceResult<EpubBook>(null, errorDescriptor: $"Unable to retrieve epub book: {ex.Message}");
+            }
+            
+        }
+
+        /// <summary>
+        /// Retrieves a current book chapter when a user is changing pages or starting a new e-book.
+        /// </summary>
+        /// <param name="epubBook">The <see cref="EpubBook"/> containing all information about a book.</param>
+        /// <returns>A <see cref="ServiceResult{TResult}"/> with how the request was handled.</returns>
+        public ServiceResult<string> GetCurrentBookChapter(int currentChapterIndex, EpubBook epubBook)
+        {
+            try
+            {
+                var currentChapter = DisplayChapterContent(currentChapterIndex, epubBook);              
+                return new ServiceResult<string>(currentChapter.Instance, errorDescriptor: "Successfully retrieved the current chapter.");
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Error retrieving the current chapter: {ex.Message}");
+                return new ServiceResult<string>(string.Empty, errorDescriptor: $"Error retrieving the current chapter: {ex.Message}.");
+            }
+        }
+
+        /// <summary>
+        /// Gets the next chapter, the interation of chapterIndex++ should happen before calling this method.
+        /// </summary>
+        /// <param name="chapterIndex">The next chapter index.</param>
+        /// <param name="epubBook">The <see cref="EpubBook"/> which needs to be read from.</param>
+        /// <returns></returns>
+        public ServiceResult<string> GetNextChapter(int chapterIndex, EpubBook epubBook)
+        {
+            try
+            {
+                var nextChapter = DisplayChapterContent(chapterIndex, epubBook);
+                return new ServiceResult<string>(nextChapter.Instance, errorDescriptor: "Successfully retrieved the current chapter.");
```

Comment:
"Successfully retrieved the *next* chapter."

Reply:
Good catch! Should be fixed. 😊